pennylane/ops/functions/equal.py

```diff
@@ -127,8 +143,10 @@ def equal_operator(op1, op2, check_interface, check_trainability, rtol, atol):
     return getattr(op1, "inverse", False) == getattr(op2, "inverse", False)
 
 
-def equal_measurements(op1, op2):
+@_equal.register
+def _equal_measurements(op1: MeasurementProcess, op2):
```

Comment:
```suggestion
def _equal_measurements(op1: MeasurementProcess, op2: MeasurementProcess):
```

Reply:
We only really *need* to specify the type hint for the first one for `singledispatch` to work correctly, but I am 100% for adding type hints whenever possible.